query/src/main/java/org/infinispan/query/dsl/embedded/impl/LuceneQueryMaker.java

```diff
@@ -115,6 +116,28 @@ public LuceneQueryMaker(SearchIntegrator searchFactory, FieldBridgeProvider<Type
          isAnalyzerRemote = true;
       }
       Query query = makeQuery(parsingResult.getWhereClause());
+
+      // an all negative top level boolean query is not allowed; needs a bit of rewriting
```

Comment:
So this means queries for example like ```-cancelled:true -delayed:true``` for hypothetical flights would not be allowed?